### PR TITLE
Libcollectdclient fix stringop warning

### PR DESCRIPTION
**Pull request by rubenk**
Changelog: libcollectdclient: fix stringop compiler warning

### CODE REVIEW

File: src/libcollectdclient/network_parse.c [hunk 1]
```diff
@@ -154,11 +154,11 @@ static int parse_string(void *payload, size_t payload_size, char *out,
                         size_t out_size) {
   char *in = payload;
 
-  if ((payload_size < 1) || (in[payload_size - 1] != 0) ||
+  if ((payload_size < 1) || (in[payload_size - 1] != '\0') ||
       (payload_size > out_size))
     return EINVAL;
 
-  strncpy(out, in, out_size);
+  strncpy(out, in, out_size - 1);
```
[the remark this answers is not in the document]
Comment (rubenk) [reply src/libcollectdclient/network_parse.c]: I check the "in" has a trailing NUL byte on line 157, so "out" is always NUL-terminated.

[the remark this answers is not in the document]
Comment (JAORMX) [reply src/libcollectdclient/network_parse.c]: @rubenk right, but now you're copying until out_size - 1, which would be before the NULL termination. so... potentially out doesn't get the NULL set.

[the remark this answers is not in the document]
Comment (octo) [reply src/libcollectdclient/network_parse.c]: I agree with @JAORMX: the function doesn't write to the last byte in `out`. That only works if `out` is zero initialized, which is not an assumption I feel comfortable with. Adding

```c
out[out_size - 1] = 0;
```

is the way to go.

File: src/libcollectdclient/network_parse.c [hunk 2]
```diff
@@ -154,11 +154,11 @@ static int parse_string(void *payload, size_t payload_size, char *out,
                         size_t out_size) {
   char *in = payload;
 
-  if ((payload_size < 1) || (in[payload_size - 1] != 0) ||
+  if ((payload_size < 1) || (in[payload_size - 1] != '\0') ||
```
Comment (octo) [inline src/libcollectdclient/network_parse.c]: Nit, at your option: Personally I prefer `0` over `'\0'`, but there is plenty of precedence for the latter.

**Pull request merged**